test/test_config.h.in

```diff
@@ -48,4 +51,110 @@ void unsetenv(const char * name)
   return;
 }
 #endif
+
+namespace sdf
+{
+  namespace testing
+  {
+    /// \brief Method to retrive root directory of project source
+    ///
+    /// This is used to get various test files
+    /// \param[inout] _dataDir Full path to the source directory
+    /// \return True if directory is set correctly, false otherwise
+    bool ProjectSourcePath(std::string &_sourceDir)
+    {
+      // Bazel builds set TEST_SRCDIR
+      if(char* dataDir = std::getenv("TEST_SRCDIR"))
+      {
+        _sourceDir = sdf::filesystem::append(dataDir, "__main__/sdformat");
+        return true;
+      }
+      else
+      {
+        // CMake builds set PROJECT_SOURCE_PATH
+        _sourceDir = PROJECT_SOURCE_PATH;
+        return true;
+      }
+
+      return false;
+    }
+
+    /// \brief Method to retrive temporary directory for test outputs
+    ///
+    /// \param[inout] _tmpDir Full path to the temp directory
+    /// \return True if directory is set correctly, false otherwise
+    bool TestTmpPath(std::string &_tmpDir)
+    {
+      // Bazel builds set TEST_UNDECLARED_OUTPUTS_DIR
+      if (const char* tmpDir = std::getenv("TEST_UNDECLARED_OUTPUTS_DIR"))
+      {
+        _tmpDir = tmpDir;
+        return true;
+      }
+
+      // Otherwise use a temporary dir relative to home
+      if (char* homeDir = std::getenv("HOME"))
```

Comment:
This results in `$HOME/tmp-sdf` being the temporary directory. Is there a way we can put it in the build directory instead?